adding kubernetes cluster installation instructions and clarifications

zoza1982 opened the pull request:
Submitting this PR per @ryane agreement with me  to document that kubernetes cluster install should be separate cluster than mantl, as otherwise it will brake existing mantl.

I made it as a separate readme doc which is easy to notice when you start aiming for kubernetes installation. Probably  this "should" ?  end up one day in official documentation. 

This contribution will also minimize  frequently asked questions on our gitter chat how to setup kubernetes as there is no existing detailed docs on it.

Let me know guys what you think....

siddharthist commented:
If we figure out a good way for playbooks to detect whether or not Mantl has been installed (vis a vis #1084), we should include that in kubernetes.yml and fail when Mesos is already present.

siddharthist commented:
Could this go in docs/ rather than in the root directory? And be linked from index.rst? I'd be happy to merge this then and make edits afterwards. Thanks for the contribution @zoza1982!

zoza1982 commented:
@siddharthist I did as proposed.  my file needs Sphinx markup to look pretty.
Let me know if new section in index.rst makes sense.

siddharthist commented:
@zoza1982 It would be best if we could move it to `docs/components/kubernetes.rst` and convert to rst format, for consistency with our other documentation (if you don't mind). And I'm not sure the new section is necessary, but it's no big deal either way. We can always take it out. Thanks for working on documentation, it's super important!

zoza1982 commented:
@siddharthist no I don't mind of course. It would not make sense not doing it. I am just not native to rst format and I have to print some cheat sheet and start doing it by example. I will start working on it.

zoza1982 commented:
@siddharthist so lets clarify my intentions:

1) Rename the file to README.rst and do the Sphinx formatting.
2) Move it to /roles/kubernetes/README.rst
3) create file docs/components/kubernetes.rst  with content of  .. include:: ../../roles/kubernetes/README.rst

That how all others are done under components.
Is this sounds good?

siddharthist commented:
@zoza1982 Sounds perfect!

zoza1982 commented:
Cool ...I am on it....

zoza1982 commented:
@siddharthist  , this is done. Please do quick review and let me know how it looks and if its ready for merge.

siddharthist commented on roles/kubernetes/README.rst:
```diff
@@ -0,0 +1,40 @@
+**REQUIREMENTS:**
+
+- Terraform binaries installed 
+- Ansible 1.9.x +  ( just run ``pip install -r requirements.txt`` ) 
```
Maybe we can shorten this to just advising users to run ``pip install -r requirements.txt``, rather than specifying an Ansible version?

siddharthist commented on roles/kubernetes/README.rst:
```diff
@@ -0,0 +1,40 @@
+- Copy ./terraform.sample.yml into a new file and populate the file (e.g terraform.yml)
+- From root folder of the project cp terraform/openstack-modules.sample.tf  .
+- Rename openstack-modules.sample.tf to openstack.tf
+- Populate your openstack.tf
```
These don't need to be specific to openstack.

siddharthist commented on roles/kubernetes/README.rst:
```diff
@@ -0,0 +1,40 @@
+Installation instructions ( openstack example ):
+------------------------------------------------
+
+**TERRAFORM STEPS:**
```
These steps actually happen after you run Terraform

siddharthist commented on roles/kubernetes/README.rst:
```diff
@@ -0,0 +1,40 @@
+
+**TERRAFORM STEPS:**
+
+- Copy ./terraform.sample.yml into a new file and populate the file (e.g terraform.yml)
```
This could probably go under "Ansible steps"

siddharthist commented on roles/kubernetes/README.rst:
```diff
@@ -0,0 +1,40 @@
+- Ansible 1.9.x +  ( just run ``pip install -r requirements.txt`` ) 
+- Fresh formed cluster with terraform on your cloud of choice
+
+Installation instructions ( openstack example ):
```
Why just Openstack? Can we make all of this more general?

siddharthist commented:
@zoza1982 I added a few comments, but this is looking really good so far! I'm excited to have a README for this component. If you're feeling up to it, you could also add a section documenting the available variables available in [the kubernetes defaults](https://github.com/CiscoCloud/microservices-infrastructure/blob/9424bbdb7f3d250722e9c3519325e0a040e5fd58/roles/kubernetes/defaults/main.yml). See [the Consul readme](https://raw.githubusercontent.com/CiscoCloud/microservices-infrastructure/master/roles/consul/README.rst) for an example. We can always add that later, as well!

zoza1982 commented:
@siddharthist  thanks for feedback.Very appreciated as I am not a big on documentation. First contribution in public so far :-) .  I will get it updated.

zoza1982 commented:
@siddharthist  alright ....one more check...I believe we should be good.

zoza1982 commented:
@siddharthist I tried to mimic consul variables doc section as much as I could .....but there is not much variables  that you wanna customize in kubernetes defaults as there is in consul.I hope it looks decent for a now.

zoza1982 commented:
@siddharthist  are we ready to merge this? Let me know if anything else needs correction. I am fully  open to criticism :-) for the benefit of improvement. Feel free to pitch in more...

siddharthist commented:
@zoza1982 LGTM :+1: It would be awesome if you could rebase this and squash all the commits into one, but nbd either way.

zoza1982 commented:
@siddharthist  ....As first time doing rebase...
I think I did what I needed to do...I rebased my kubernetes-setup notes into my master.
And its all now here....  https://github.com/zoza1982/microservices-infrastructure.git and deleted my branch . 

I am not sure what I am supposed to do now...do I make new pull request from my master and reference this?

siddharthist commented:
@zoza1982 [Here's a good tutorial](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history). What I was talking about in particular is covered under "Squash commits together". I'll just make a pull request into your branch, though, so don't worry about it. Sorry, didn't realize that it would be complicated!
